Modules/@babylonjs/react-native-basekit/scripts/install.js

```diff
@@ -0,0 +1,174 @@
+const https = require('https');
+const zlib = require('zlib');
+const tar = require('tar');
+const path = require('path');
+const fs = require('fs');
+
+function getCachePathCandidates() {
+  return [
+    process.env.npm_config_sass_binary_cache,
+    process.env.npm_config_cache,
+  ].filter(function(_) { return _; });
+}
+
+function getCached(package, binaryFilename) {
+  return [getCachePathCandidates(), 'BabylonReactNative', package.version, binaryFilename].join('/');
+}
+
+function useCachedFile(filePath) {
+  if (fs.existsSync(filePath)) {
+    console.log(`Using cached file ${filePath}`);
+    const gunzip = zlib.createGunzip();
+    const untar = tar.extract({ cwd: process.cwd() }); // Extract to the current working directory
+
+    const sourceStream = fs.createReadStream(filePath); 
+    sourceStream
+    .pipe(gunzip) // Decompress the .tar.gz file
+    .pipe(untar) // Extract the tar contents to the cwd
+    .on('error', (err) => {
+      console.error('Error extracting .tar.gz file:', err);
+      process.exit(1);
+    })
+    .on('end', () => {
+      console.log('Extraction completed successfully.');
```

Comment:
this will happen async. as you don't define a callback to be executed when done, you can't 100% be sure when this has happened.
The process will stay working until reaching this point (or until and error occurs) but it might be good to catch them either using success/fail callbacks or wrapping them in a promise

Reply:
Reworked as well.